rita/src/rita_common/dao_manager/mod.rs

```diff
@@ -0,0 +1,142 @@
+// Manages subnet DAO membership, DAOManager mantains a cache of subnet DAO entries.
+// The workflow goes as follows, an actor message DAOCheck is sent to DAOManager
+// if the identity is not on the DAO it will run a callback to tunnel manager to remove
+// that tunnel from operation. If the identity is on the DAO it will do nothing.
+// Entires from the DAO are cached for a configurable amount of time, this may of course
+// have the effect of adding someone to the DAO taking time to kick in.
+
+extern crate rustc_hex;
+
+use actix::prelude::*;
+use std::collections::HashMap;
+use std::net::IpAddr;
+use std::net::Ipv6Addr;
+use std::time::Instant;
+use tokio::reactor::Handle;
+use web3::contract::Contract;
+use web3::contract::Options;
+use web3::transports;
+use web3::Web3;
+
+use althea_types::EthAddress;
+use althea_types::Identity;
+use settings::RitaCommonSettings;
+
+use SETTING;
+
+pub struct DAOManager {
+    entries: HashMap<Identity, Vec<DAOEntry>>,
+}
+
+impl Actor for DAOManager {
+    type Context = Context<Self>;
+}
+impl Supervised for DAOManager {}
+impl SystemService for DAOManager {
+    fn service_started(&mut self, _ctx: &mut Context<Self>) {
+        info!("Tunnel manager started");
+    }
+}
+
+impl Default for DAOManager {
+    fn default() -> DAOManager {
+        DAOManager::new()
+    }
+}
+
+impl DAOManager {
+    fn new() -> DAOManager {
+        DAOManager {
+            entries: HashMap::<Identity, Vec<DAOEntry>>::new(),
+        }
+    }
+}
+
+pub struct DAOEntry {
+    on_list: bool,
+    dao_address: EthAddress,
+    id: Option<Identity>,
+    last_updated: Instant,
+}
+
+pub struct DAOCheck(Identity);
+impl Message for DAOCheck {
+    type Result = ();
+}
+
+impl Handler<DAOCheck> for DAOManager {
+    type Result = ();
+
+    fn handle(&mut self, msg: DAOCheck, _: &mut Context<Self>) -> Self::Result {
+        let their_id = msg.0;
+    }
+}
+
+/// True if timestamp does not need to be updated
+fn timer_check(timestamp: Instant) -> bool {
+    if Instant::now() - timestamp < SETTING.get_dao().cache_timeout {
```

Comment:
Could be simplified as just `Instant::now() - timestamp < SETTING.get_dao().cache_timeout`